pkg/querier/blocks_store_queryable.go

```diff
@@ -139,7 +139,7 @@ func NewBlocksStoreQueryableFromConfig(querierCfg Config, gatewayCfg storegatewa
 
 	if gatewayCfg.ShardingEnabled {
 		storesRingCfg := gatewayCfg.ShardingRing.ToRingConfig()
-		storesRingBackend, err := kv.NewClient(storesRingCfg.KVStore, ring.GetCodec())
+		storesRingBackend, err := kv.NewClient(storegateway.RingNameForClient+"-block-store", storesRingCfg.KVStore, ring.GetCodec(), reg)
```

Comment:
The generated name here looks a bit weird to me. Maybe `querier-store-gateway` may be more clear?

Reply:
Sounds good, I wasn't sure how best to name some of these clients.